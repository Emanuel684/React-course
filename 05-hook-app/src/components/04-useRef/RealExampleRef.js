import React, { useState } from 'react';
import '../02-useEffect/effects.css';
import MultipleCustomsHooks from '../03-examples/MultipleCustomsHooks';

const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>RealExampleREf</h1>
            <hr />

            {show && <MultipleCustomsHooks />}

            <button className="btn btn-primary mt-5" onClick={() => {
                setShow(!show);
            }}>
                Show/Hide
            </button>

        </div>
    )
};

export default RealExampleRef;