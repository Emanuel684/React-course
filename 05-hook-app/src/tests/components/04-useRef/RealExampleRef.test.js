import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import RealExampleRef from '../../../components/04-useRef/RealExampleRef';

describe('Prueba en el componente de <RealExampleRef />', () => {

    const wrapper = shallow(<RealExampleRef />);

    test('Debe de mostrar correctamente el componente de <RealExampleRef />', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar el componente <MultipleCustomsHooks />', () => {

        wrapper.find('button').simulate('click');

        expect( wrapper.find('MultipleCustomsHooks').exists() ).toBe(true);

    });
    

});