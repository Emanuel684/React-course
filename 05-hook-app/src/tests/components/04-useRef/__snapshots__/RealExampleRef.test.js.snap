// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente de <RealExampleRef /> Debe de mostrar correctamente el componente de <RealExampleRef /> 1`] = `
<div>
  <h1>
    RealExampleREf
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
